src/helpers/sendBinEmbed.ts

```diff
@@ -1,20 +1,53 @@
-import { Message, MessageEmbed, MessageReaction, User } from "discord.js";
+import {
+	BufferResolvable,
+	Collection,
+	Message,
+	MessageAttachment,
+	MessageEmbed,
+	MessageReaction,
+	Snowflake,
+	User,
+} from "discord.js";
+import { Stream } from "stream";
 
-const noop = (): void => {};
+const noop = (): undefined => undefined;
+
+const MAX_FILE_SIZE = 8 * 1024 * 1024; // 8MiB
 
 export async function sendBinEmbed(
 	message: Message,
 	description: string,
 	extender?: (embed: MessageEmbed) => MessageEmbed,
+	attachments?: Collection<Snowflake, MessageAttachment>,
 ): Promise<void> {
 	const embed = new MessageEmbed({ description })
 		.setAuthor(message.member!.displayName, message.author.displayAvatarURL({ dynamic: true }))
 		.setTimestamp(message.createdAt);
+
 	if (extender) {
 		extender(embed);
 	}
 
-	const botMessage = await message.channel.send(embed).catch(noop);
+	const waitMessage = await message.channel.send("Transformation du message en cours...").catch(noop);
+	const files: (BufferResolvable | Stream)[] = [];
+
+	if (attachments) {
+		let totalSize = 0;
+
+		attachments.reduce((acc, cur) => {
+			if (totalSize > MAX_FILE_SIZE) {
+				return acc;
+			}
+
+			acc.push(cur.attachment);
+			totalSize += cur.size;
+			return acc;
+		}, files);
+	}
```

Comment:
```suggestion
	const files = attachments.reduce((attachment) => attachment < MAX_FILE_SIZE);
```

Reply:
Ça c'est pas un reduce valide...

Reply:
Là tu me prends les fichiers inférieurs à la taille, or, même si Discord dit l'inverse, il limite à 8Mo *au total*

Reply:
@Aakodal j'ai édité la suggestion entre temps.

Reply:
mauvais typage, la fallback value de ta ternaire envoie un array de MessageAttachment, toi tu attends une collection.

Reply:
Résolus.

Reply:
ouais laisse, je passe sur un for of, cf mon commit qui arrive d'ici quelques secondes